### ISSUE TITLE
setAllPerms<-.how removes the allPerms class from all.perms, causing error on subsequent calls to check()

### ISSUE THREAD

**Issue by brendanf** [gavinsimpson/permute#16]
The following code fails:
```
perm <- how()
perm <- check(4, control=perm)$control
check(4, perm)
#Error in check(4, perm) : 'control$all.perms' must be of class 'allPerms'.
```
The problem is setFoo-methods.R line 70-71:
```
if (!is.null(value))
        value <- as.matrix(value)
```
which strips the 'allPerms' class.

**Comment by gavinsimpson** [comment]
Thanks @brendanf. I have a fix that I'm testing at the moment.

**Comment by gavinsimpson** [comment]
This is now fixed in the github version via ce4e935, a77a036, and 1e01040
